### DISCUSSION TITLE
Proposal: <note> documentation element

### DISCUSSION THREAD

**Discussion by sharwell** [dotnet/csharplang#8972] [Language Ideas]
Champion issue: https://github.com/dotnet/csharplang/issues/8973

I propose updating the C# language specification to define a `<note>` block-level element in the context of XML documentation comments. The `<note>` element has a `type` attribute, which is allowed to be any identifier but the following note types are predefined:

* `note`: A general note. This is the default if the `type` attribute is omitted.
* `implement`: Note to implementers
* `caller`: Note to callers
* `inherit`: Note to inheritors
* `important`: An important note
* `security`: A security note

🔗 [note (Sandcastle XML Comments Guide)](http://ewsoftware.github.io/XMLCommentsGuide/html/4302a60f-e4f4-4b8d-a451-5f453c4ebd46.htm)

**Comment by SilentSin** [comment]
I like it. In the absence of any stricter enforcement, it would be useful to tell a child class whether a method override should call the base or not without any callers caring about that aspect when they read the comment.

Should `override` be separate from `inherit`? I can't see any ambiguity, but it seems like it might be appropriate.
